search.py

```diff
@@ -647,19 +647,21 @@ def handle_get_user_profile(task: daemon_tasks.Task):
 
 
 def query_to_url(query={}, params={}):
-    # build a new request
+    """Build a new search request by parsing query dictionaty into appropriate URL.
+    Also modifies query and params and adds some stuff in there which is very misleading anti-patter.
+    TODO: just convert to URL here and move the sorting and adding of params to separate function.
+    https://www.blenderkit.com/api/v1/search/
+    """
     url = f"{paths.BLENDERKIT_API}/search/"
 
-    # build request manually
-    # TODO use real queries
     requeststring = "?query="
-    #
     if query.get("query") not in ("", None):
-        requeststring += query["query"]  # .lower()
-    for i, q in enumerate(query):
+        requeststring += urllib.parse.quote_plus(query["query"])  # .lower()
```

Comment:
This probably might be added to the `for` loop, I don't think that server minds if `query` is the first parameter. Only the resulting string would need to be constructed by `"+".join()`.

Then the whole `urllib.parse.quote_plus(str(query["query"]))` could be set to variable once.

But there might be some unexpected problems with that, so maybe that could wait for the refactor.